static/js/src/advantage/react/components/Subscriptions/SubscriptionDetails/SubscriptionDetails.test.tsx

```diff
@@ -13,39 +16,90 @@ describe("SubscriptionDetails", () => {
 
   beforeEach(async () => {
     queryClient = new QueryClient();
-    contract = freeSubscriptionFactory.build();
+    contract = userSubscriptionFactory.build();
     queryClient.setQueryData("userSubscriptions", [contract]);
   });
 
   it("initially shows the content", () => {
     const wrapper = mount(
       <QueryClientProvider client={queryClient}>
-        <SubscriptionDetails onCloseModal={jest.fn()} />
+        <SubscriptionDetails onCloseModal={jest.fn()} selectedToken="0" />
       </QueryClientProvider>
     );
     expect(wrapper.find("DetailsContent").exists()).toBe(true);
     expect(wrapper.find("SubscriptionEdit").exists()).toBe(false);
   });
 
-  // TODO: remove skip from these tests when the subscription details can load
-  // non-free subscriptions.
-  it.skip("can show the edit form", () => {
+  it("can show the edit form", () => {
     const wrapper = mount(
       <QueryClientProvider client={queryClient}>
-        <SubscriptionDetails onCloseModal={jest.fn()} />
+        <SubscriptionDetails onCloseModal={jest.fn()} selectedToken="0" />
       </QueryClientProvider>
     );
     wrapper.find("Button[data-test='edit-button']").simulate("click");
     expect(wrapper.find("SubscriptionEdit").exists()).toBe(true);
     expect(wrapper.find("DetailsContent").exists()).toBe(false);
   });
 
-  // TODO: remove skip from these tests when the subscription details can load
-  // non-free subscriptions.
-  it.skip("disables the buttons when showing the edit form", () => {
+  it("closes the edit form when changing subscriptions", () => {
+    queryClient.setQueryData("userSubscriptions", [
+      userSubscriptionFactory.build(),
+      userSubscriptionFactory.build(),
+      freeSubscriptionFactory.build(),
+    ]);
+    // Create a wrapping component to pass props to the inner
+    // SubscriptionDetails component. This is needed so that setProps will
+    // update the inner component.
+    const ProxyComponent = ({ selectedToken }: { selectedToken: string }) => (
+      <QueryClientProvider client={queryClient}>
+        <SubscriptionDetails
+          onCloseModal={jest.fn()}
+          selectedToken={selectedToken}
+        />
+      </QueryClientProvider>
+    );
+    const wrapper = mount(<ProxyComponent selectedToken="0" />);
+    wrapper.find("Button[data-test='edit-button']").simulate("click");
+    expect(wrapper.find("SubscriptionEdit").exists()).toBe(true);
+    expect(wrapper.find("DetailsContent").exists()).toBe(false);
+    wrapper.setProps({ selectedToken: "1" });
```

Comment:
Maybe add a comment here. It's not clear that setting this changes selected subscription.

Reply:
Good point, done.